luarules/gadgets/unit_maverick_exp.lua

```diff
@@ -20,39 +20,42 @@ local GetUnitExperience = Spring.GetUnitExperience
 local SetUnitMaxRange = Spring.SetUnitMaxRange
 local GetUnitDefID = Spring.GetUnitDefID
 
-local isMaverick = {}
-for udid, ud in pairs(UnitDefs) do
-	if string.find(ud.name, 'armmav') then
-		isMaverick[udid] = true
-	end
-end
-local maverickOriginalRange = WeaponDefNames["armmav_armmav_weapon"].range
-
 local updateList = {}
 
 function gadget:GameFrame(n)
 	for unitID in pairs(updateList) do
 		local curExp = GetUnitExperience(unitID)
         if curExp then
-            local limExp = ((3*curExp)/(1+3*curExp))*0.6
-            local newRange = maverickOriginalRange * ( 1 + limExp )
+			local unitDef = UnitDefs[GetUnitDefID(unitID)]
+
+			local rangeXPScale = unitDef.customParams.rangexpscale
+            local limExp = ((3*curExp)/(1+3*curExp))*rangeXPScale
+
+            local newRange = WeaponDefs[unitDef.weapons[1].weaponDef].range  * ( 1 + limExp )
+
             SetUnitWeaponState(unitID, 1, "range", newRange)
             SetUnitMaxRange(unitID,newRange)
         end
     end
 	updateList = {}
 end
 
-function GG.requestMaverickExpUpdate(unitID)
-	if not isMaverick[GetUnitDefID(unitID)] then
+function GG.requestMaverickExpUpdate(unitID)	
+	if (not(string.find(UnitDefs[attackerDefID].name, "armmav"))) then
```

Comment:
Rather than do a string match on the unit's name, check if it has the `rangexpscale` customparam. String matching on the unit name is fragile and will break for other units.

Reply:
Have solved this :)